app/waf/src/services/network.service.ts

```diff
@@ -51,23 +52,27 @@ export class NetworkDriver {
   }
 
   async createPort(
-    url: string,
     userToken: string,
     portParams: PortCreationParams,
   ): Promise<string> {
     try {
       // TODO: exception check.
       return await this.networkEndpoint(portParams.regionName)
-        .then(networkUrl => {
-          let portCreationBody: PortCreationRequestBody = {
-            port: {network_id: portParams.networkId},
+        .then(async networkUrl => {
+          let url = networkUrl + '/v2.0/ports';
+          let body: PortsRequest = {
+            port: {
+              network_id: portParams.networkId,
+            },
           };
 
-          return this.networkService.v2CreatePort(
-            networkUrl,
-            userToken,
-            portCreationBody,
-          );
+          if (portParams.fixedIp) {
+            body.port.fixed_ips = [{ip_address: portParams.fixedIp}];
+          }
+
+          body.port.name = 'f5-' + portParams.name;
```

Comment:
Will this port name be "f5-adcId-" ? Or something else?

Reply:
Right. The `portParams.name` should be assembled with adc ID in caller context. May probably `<adcId>-<type>-<name>`.  
adcId is the adc ID in DB. 
type is the type of port 'mgmt' 'ha' 'ext' 'int'. 
name is optional: 

```
  networks: {
    name?: string;                <- name 
    type: 'mgmt' | 'ext' | 'int' | 'ha';   <- type
    networkId: string;
    fixedIp?: string;
    //floatingIp?: string;
    portId?: string; // cannot be appointed.
    ready?: boolean; // cannot be appointed.
    //vips?: [string]; // cannot be appointed.
  }[];
```